### PR TITLE
feat: tensorflow frontend `avg_pool_2d`

### LINKED ISSUE #22783: avg_pool2d

### PR DESCRIPTION
**Pull request by kahina45678**
# PR Description 
this is only a partial solution. i tried to implement the function's logic first.

## Related Issue 

Close #22783

## Checklist 

- [x] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you follow the steps we provided?

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]
Hello @kahina45678 , thanks for the contribution
the function you are trying to implement should be implemented in this file `ivy/functional/frontends/tensorflow/nn.py`, also please read the contribution guide here https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends.html and check the similar implemented functions in the same open tast

**Comment by kahina45678** [comment]
hello @MahmoudAshraf97 !
how about now ? what do i need to add or change ? Thank you.

File: ivy/functional/frontends/tensorflow/nn.py
Comment (MahmoudAshraf97) [inline ivy/functional/frontends/tensorflow/nn.py:1]: we must strictly use ivy functions in the frontends, and also delete the old file as it's not needed anymore

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**